hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/ExtensionEventListener.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.api.services;
+
+import java.util.TreeSet;
+
+import org.hawkular.alerts.api.model.event.Event;
+
+
+/**
+ * A listener that will process received Events.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+public interface ExtensionEventListener {
+
+    /**
+     * Process Events received in the engine before it is processed by the Alerting Engine.
+     *
+     * @param events the Events received by the engine before to be processed
+     * @return the resulting Events processed by the extension and sent to the engine
+     */
+    TreeSet<Event> processEvents(TreeSet<Event> events);
```

Comment:
The suggestions for the javadoc are good.
I will incorporate them into the review, perhaps trying to unify both proposals.